core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -1390,6 +1411,30 @@ extern "C" fn new_show_progress(n_args: usize, args: *const Obj, kwargs: *mut Ma
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_show_progress_coinjoin(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let indeterminate: bool = kwargs.get_or(Qstr::MP_QSTR_indeterminate, false)?;
+        let time_ms: u32 = kwargs.get_or(Qstr::MP_QSTR_time_ms, 0)?;
+        let skip_first_paint: bool = kwargs.get_or(Qstr::MP_QSTR_skip_first_paint, false)?;
+
+        // The second type parameter is actually not used in `new()` but we need to
+        // provide it.
+        let progress = CoinJoinProgress::<_, Never>::new(title, indeterminate);
+        let obj = if time_ms > 0 && indeterminate {
+            let timeout = Timeout::new(time_ms);
+            LayoutObj::new((timeout, progress.map(|_msg| None)))?
```

Comment:
For understandabilty purposes, would it make sense to rename `Component::map` and `pub struct Map` to `map_event` and `EventMap` or something like that to signify it is applying the map only to the event?

Reply:
Not sure if it's ever useful to applying map to events but if you think it helps I've renamed the struct to `MsgMap`. I'd like to keep the method name, overloading the name seems to be a common practice: https://doc.rust-lang.org/std/index.html?search=map

ffa5483099d9e1902661fddcbeb52826e681170c

Reply:
Agree with keeping `map`, thanks for renaming `MsgMap`, it was confusing me a little bit